gobblin-data-management/src/test/java/org/apache/gobblin/data/management/copy/iceberg/IcebergDatasetTest.java

```diff
@@ -176,22 +176,72 @@ public void testGenerateCopyEntitiesMultiSnapshotWhenDestEmpty() throws IOExcept
     verifyCopyEntities(copyEntities, expectedPaths);
   }
 
+  @Test
+  public void testFsOwnershipAndPermissionPreservationWhenDestEmpty() throws IOException {
+    List<String> expectedPaths = Arrays.asList(METADATA_PATH, MANIFEST_LIST_PATH_0,
+        MANIFEST_PATH_0, MANIFEST_DATA_PATH_0A, MANIFEST_DATA_PATH_0B);
+
+    MockFileSystemBuilder sourceBuilder = new MockFileSystemBuilder(SRC_FS_URI);
+    sourceBuilder.addPaths(expectedPaths);
+    FileSystem sourceFs = sourceBuilder.build();
```

Comment:
for robust testing, shouldn't we set particular permissions and ownership for the parent/ancestors of at least some paths?  I'd hope at least two levels get differing values, to truly demonstrate that the recursive walk toward root accurately preserves what it finds (in the correct order).

BTW, could always check whether there may be a test on the hive distcp side to take ideas from...